### PR TITLE
RavenDB-16902 - Document with unicode character should not have changes on load

### PR DESCRIPTION
**Pull request by shiranshalom**

### CODE REVIEW

File: src/Raven.Client/Json/BlittableOperation.cs [hunk 1]
```diff
@@ -159,6 +161,37 @@ public static bool EntityChanged(BlittableJsonReaderObject newObj, DocumentInfo
...
+
+            if (oldProp.Value is LazyStringValue lsv)
+            {
+                int pos = lsv.Size;
+                int numOfEscapePositions = BlittableJsonReaderBase.ReadVariableSizeInt(lsv.Buffer, ref pos);
+                if (numOfEscapePositions == 0)
+                    return false;
+
+                var memoryStream = new MemoryStream();
+                using (var textWriter = new AsyncBlittableJsonTextWriter(context, memoryStream, CancellationToken.None))
```
Comment (ppekrol) [inline src/Raven.Client/Json/BlittableOperation.cs]: why Async? there is a `BlittableJsonTextWriter`

Comment (shiranshalom) [reply src/Raven.Client/Json/BlittableOperation.cs]: Done

File: src/Raven.Client/Json/BlittableOperation.cs [hunk 2]
```diff
@@ -167,28 +167,29 @@ public static bool EntityChanged(BlittableJsonReaderObject newObj, DocumentInfo
...
-            fixed (byte* pBuff = bytes.Array)
-            {
-                // need to ignore the quote marks
-                using var str = context.AllocateStringValue(null, pBuff + bytes.Offset + 1, (int)memoryStream.Length -2);
+                int pos = lsv.Size;
+                int numOfEscapePositions = BlittableJsonReaderBase.ReadVariableSizeInt(lsv.Buffer, ref pos);
+                if (numOfEscapePositions == 0)
+                    return false;
+
+                var memoryStream = new MemoryStream();
```
Comment (garayx) [inline src/Raven.Client/Json/BlittableOperation.cs]: using

Comment (shiranshalom) [reply src/Raven.Client/Json/BlittableOperation.cs]: Done

**Review by ayende — APPROVED** [review]

**Pull request merged**